src/coord/src/coord.rs

```diff
@@ -539,7 +539,7 @@ where
         // Cleanly drain any pending messages from the worker before shutting
         // down.
         drop(internal_cmd_tx);
-        while let Some(_) = block_on(messages.next()) {}
+        while block_on(messages.next()).is_some() {}
```

Comment:
is this because of some new clippy lint?

My preference would be to just disable that lint - useless clippy lints are my least favorite thing about changing Rust versions.

Reply:
Yeah I don't particularly love this lint, though I don't hate it. I'm going to leave as is because technically we've had this lint on for a while and it was just a bug in Clippy that it didn't apply to `while let`, just `if let`. But I'd be more than happy if a PR to disable it flew past.